cpp/src/arrow/util/vector.h

```diff
@@ -78,8 +78,8 @@ std::vector<T> ReplaceVectorElement(const std::vector<T>& values, size_t index,
 
 template <typename T, typename Predicate>
 std::vector<T> FilterVector(std::vector<T> values, Predicate&& predicate) {
-  auto new_end =
-      std::remove_if(values.begin(), values.end(), std::forward<Predicate>(predicate));
+  auto new_end = std::stable_partition(values.begin(), values.end(),
```

Comment:
I'm curious, is there any reason not to use `remove_if`?

Reply:
We want `keep_if` not `remove_if` though I suppose we could wrap `predicate` and invert it.

Reply:
Follow up at #12949